### PR TITLE
[metrics] standardize/simplify export pipeline setup

### LINKED ISSUE #343: Add simplified constructor for simple metric SDK setup: stdout exporter.
It takes about 15 lines of code to setup an exporter. This is because the complete export pipeline has a number of various configurations possible, but it confuses most users who do not care to change the standard configuration.

Each exporter should provide a simple method to set it up using a recommended setup. For example, the stdout exporter should by default (IMO) use: the inexpensive measure option, the ungrouped batcher (stateful), and a push controller, e.g., the code snippet below.

```
	selector := simple.NewWithInexpensiveMeasure()
	exporter, err := metricstdout.New(options)
	if err != nil {
		return nil
	}
	batcher := ungrouped.New(selector, metricsdk.NewDefaultLabelEncoder(), true)
	pusher := push.New(batcher, exporter, time.Second)
	pusher.Start()
         return pusher
```

### PR DESCRIPTION
**Pull request by matej-g**
This PR implements a method for simplified export pipeline setup for stdout exporter.

Resolves #343

Co-authored-by: Liz Fong-Jones <lizf@honeycomb.io>

### CODE REVIEW

File: exporter/metric/stdout/stdout.go [hunk 1]
```diff
@@ -98,6 +101,20 @@ func New(options Options) (*Exporter, error) {
 	}, nil
 }
 
+// NewExportPipeline sets up a complete export pipeline with the recommended setup
```
Comment (jmacd) [inline exporter/metric/stdout/stdout.go]: Looks good. Will you add a comment here explaining the standard usage is something like

```
   metricSDK, err := stdout.NewExportPipeline(stdout.Options{...})
   if err != nil {
       ...
    }
   defer metricSDK.Stop()

   global.SetMeterProvider(metricSDK)

   // ... Done!
```

Comment (paivagustavo) [reply exporter/metric/stdout/stdout.go]: Considering that some exporters should be created with specific `selectors`, `batchers` and so on, should we standardize that only the `NewExportPipeline` is exported on all supported exporters? (and maybe rename it to `New()`?) 

Is there an use for exposing the `stdout.New()` with this new additions? Or should all customization be done through options? 

I'm asking this because users could misuse exporters by only creating an `stdout.New()`, for example, and thinking that this is enough to have an exporter configured. Or even creating invalid configurations.

Comment (jmacd) [reply exporter/metric/stdout/stdout.go]: Maybe we should avoid `New` altogether and export two functions: `NewPipeline()` to build and start a pipeline, `NewExporter()` to return a plain exporter that could be explicitly configured.

Comment (rghetia) [reply exporter/metric/stdout/stdout.go]: can we also set the provider in NewPipeline(). So it would look
```go
pipeline, err := stdout.NewPipeline(stdout.Options{...})
if err != nil {
 ...
}
defer pipeline.Stop()
// ... Done
```

Comment (jmacd) [reply exporter/metric/stdout/stdout.go]: I would probably rename the method if it also sets the global provider. Maybe `Install()`

**Comment by jmacd** [comment]
How about two methods, one `InstallNewPipeline` and one `NewPipeline`.  The `Install` form calls `NewPipeline` and then sets the global provider.
```
pipeline, err := stdout.InstallNewPipeline(stdout.Options{...})
if err != nil {
 ...
}
defer pipeline.Stop()
// ... Done
```

File: exporter/metric/stdout/stdout.go (hunk 1 above)
Comment (lizthegrey) [reply exporter/metric/stdout/stdout.go]: Pushed commit that addresses this.

**Comment by lizthegrey** [comment]
I've done as you asked on behalf of the original pull submitter, PTAL.

File: exporter/metric/internal/statsd/conn.go [hunk 2]
```diff
@@ -34,8 +34,8 @@ import (
 )
 
 type (
-	// Config supports common options that apply to statsd exporters.
-	Config struct {
+	// Options supports common options that apply to statsd exporters.
+	Options struct {
```
Comment (jmacd) [inline exporter/metric/internal/statsd/conn.go]: I think this is an unrelated remark.

I thought we had moved toward the practice of using "Config" as the structure name, and Option as a functional argument (`func(*Config)`), and Options as a `[]Option`.  See api/trace `StartConfig` and `StartOption`, for example.

That's why I prefer this struct be called Config.

Comment (jmacd) [reply exporter/metric/internal/statsd/conn.go]: See https://github.com/open-telemetry/opentelemetry-go/pull/369

**Comment by lizthegrey** [comment]
> I like the new simplified exporter setup. I would prefer to keep a `Config` struct.

Alright. Let's merge this as is (to make the one case that's Options go to Config for consistency), and then I'll do a quick followup to rename all the Options systematically to Config.

File: exporter/metric/internal/statsd/conn.go (hunk 2 above)
Comment (lizthegrey) [reply exporter/metric/internal/statsd/conn.go]: The goal of this was to standardize the patterns for all Exporter options/config. Prometheus and stdout were using options, statsd was using config...

**Comment by lizthegrey** [comment]
@paivagustavo want to take a look before we merge?
